synthesis/codon/analyzer_test.go

```diff
@@ -0,0 +1,57 @@
+package codon
+
+import (
+	"testing"
+
+	"github.com/TimothyStiles/poly/io/genbank"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+func TestComputeStats(t *testing.T) {
```

Comment:
Unit tests should not rely on file I/O unless absolutely necessary. These are already table-driven tests, the `tests` struct should be of the following form:

```
struct {
    name string

    input genbank.Genbank
    want *Stats
}
```

Reply:
I was basing it off the existing [codon test](https://github.com/TimothyStiles/poly/blob/main/synthesis/codon/codon_test.go#L80) which read the genebank files, I assume that's because the genebank input you'd need to include in your table test would be rather large? I can change it to the object if we're ok with it

Reply:
@Koeng101 I guess this just brings up my question again as to whether we want to handle genbank objects or just string sequences.

I totally forgot that the `genbank.Genbank` is a massive struct - I'm ok with reading it from file, you're right that it would be annoying to type out.

Reply:
`genbank.Genbank` structs are silly big, so I think we should just read from a file for tests, though for examples I think we should always have it as a string (so it can run on the docs website well).

The reason the tests here are off of genbank files is because that is where pretty much all the codon tables will come from until we implement #247 , which also requires quite a bit of engineering effort, I think.  If we do it off of just sequences, for example, you can miss a lot of the small things that matter when parsing a real codon table from a real file.

To put #247 in perspective, the codon usage database, the most widely used database for codon usage, was last updated in [2007](https://www.kazusa.or.jp/codon/). So there is lots of room for us to improve over the state-of-the-art.